packages/core/core/src/utils/runtimeCache/index.ts

```diff
@@ -0,0 +1,32 @@
+import { useVSFContext, sharedRef } from '../';
+import { Ref } from '@vue/composition-api';
+
+const _isContentCached = (content) => {
+  if (!content) return false;
+  if (Array.isArray(content)) return content.length;
+  if (typeof content === 'object') return Object.keys(content).length;
+  return true;
+};
+
+export const setCacheTimestamp = (key: string) => {
+  const { $sharedRefsMap } = useVSFContext();
+  const existingTimestamp = $sharedRefsMap.get(key);
+  if (existingTimestamp) return existingTimestamp;
+  const timestamp = sharedRef(Date.now(), key);
+  return timestamp;
+};
+
+export const isCacheValid = (
+  content: any,
+  timestamp: Ref<number>,
+  cacheTimeToLive?: number
```

Comment:
I would keep it in MS, not seconds

Reply:
Updated in [c054d61](https://github.com/vuestorefront/vue-storefront/pull/6420/commits/c054d61b40e7f812f9d6d5c4ab3cc7ce4be7ec8c).

Reply:
But isn't re-calculating & passing such large numbers as arguments a little bit too cumbersome?